pkg/controller/certificates/issuing/issuing_controller.go

```diff
@@ -236,6 +236,24 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return nil
 	}
 
+	certIssuingCond := apiutil.GetCertificateCondition(crt, cmapi.CertificateConditionIssuing)
+	crReadyCond := apiutil.GetCertificateRequestCondition(req, cmapi.CertificateRequestConditionReady)
+	if certIssuingCond == nil {
+		// This should never happen
+		log.V(logf.ErrorLevel).Info("Certificate does not have an issuing condition")
+		return nil
+	}
+	// If the CertificateRequest for this revision failed before the
+	// Issuing condition was last updated on the Certificate, then it must be a
+	// failed CertificateRequest from the previous issuance for the same
+	// revision. Leave it to the certificate-requests controller to delete the
+	// CertificateRequest and create a new one.
+	if req.Status.FailureTime != nil &&
+		req.Status.FailureTime.Before(certIssuingCond.LastTransitionTime) && crReadyCond.Reason == cmapi.CertificateRequestReasonFailed {
```

Comment:
I'm pretty nervous about this time comparison.. I don't think these fields are robust enough to use as part of the state machine. Why can't we inspect the actual `revision` on the failed CertificateRequest (stored as an annotation) to determine whether it's from a previous issuance?

As an example of how this could trip up, if two requests fail in a row then the `LastTransitionTime` *won't* be updated as the condition's state has not *transitioned*.

Would utilising the actual `revision` annotation work for this?